---
layout: page
organizer: Sheffield Bioinformatics Core
category: workshop
permalink: /training/2024_06_12_cytoscape
title: Analysis of Biological Networks using Cytoscape
excerpt: "Analysis of Biological Networks using Cytoscape"
startDate: 2024-06-12
date: 2024-06-12
endDate: 2024-06-12
end-date: 2024-06-12
startTime: 09.30
from: "09:30"
endTime: 16.30
to: "16:30"
city: Sheffield, or online
postcode: S3 7ND
location: Bartolome House, Seminar Room EG03, or ONLINE
country: United Kingdom
venue: Bartolome House, Seminar Room EG03, or ONLINE
description: "Course under development...."
keywords: bioinformatics, NGS, pathways, networks
difficulty: beginner
contact: bioinformatics-core@sheffield.ac.uk
speaker: Dr. Akshay Bhat
---

## Workshop Information

- Wednesday 12th June 09:30 - 16:30 (approx)
- Bartolome House, Seminar Room EG03 or ONLINE

## Booking Link

Please book your place using [this link](https://onlineshop.shef.ac.uk/conferences-and-events/faculty-of-medicine-dentistry-and-health/medical-school)



## Overview

The workshop is designed to give you an introduction to using [*Cytoscape*](https://cytoscape.org/), which is a network-based software application to perform bioinformatics data and pathway enrichment analysis. The particular use-case in this course is that of RNA-seq data, but many of the concepts are used for other types of omics-analysis. 

## Who should attend this course?

Researchers in life sciences who want to get an appreciation for the computational network biology steps involved in RNA-seq analysis, and how to execute best-practice Cytoscape workflows.

## Objectives:- After this course you should be able to:

- Have more detailed understanding of terms used in network biology.
- Use Cytoscape to better visualise enriched terms from an RNA-seq dataset
- Know what tools are available in App Store for RNA-seq, miRNA data and pathway analysis.


## Aims:- During this course you will learn about:


- Cytoscape and specific plugins used for RNA-Seq data analysis.
- Sources for integrating differentially expressed genes within pathway modules.
- Hierarchical clustering using String and FunctionalEnrichment apps.

# Software installation

You will need to bring an internet-enabled laptop to the course and install the latest versions of Cytoscape before coming to the course

1) Install Cytoscape
  - Go to [Cytoscape](https://cytoscape.org/download.html) and click Download

2) Install the Cytoscape packages/Apps required for the course

  - Go to [ClueGO](https://apps.cytoscape.org/apps/cluego) and [CluePedia](https://apps.cytoscape.org/apps/cluepedia)
  - Click Download or Install (if your cytoscape application is already open)
  - More on installation, documentation and how to integrate the package with R or Python is described here:

3) Download and unzip this file containing the data for the course

- [Link to Data](https://github.com/a1aks/Cytoscape_Course/tree/0b870086afbc1b678f10f9e9bcf55ec342494660/Data_Files)

4) (Optional) Watch these short introductory videos

- [A gentle introduction to Cytoscape](https://www.youtube.com/watch?v=YH-XXHhrv58)
- [Visualisation of Biological Networks using Cytoscape](https://www.youtube.com/watch?v=sqM7ZLhTJoM&list=PLFQS98nmv__zaNXLUQACQu1dHWjtpToKN&index=3)


## Timetable (Provisional)

- Session 1 Wednesday 12th June 9:30 - 13:00
  + Importing example cancer RNA-seq counts into Cytoscape
  + Querying pathway and interaction sources using Cytoscape - App.
  + Styling the network and saving for publication.
- Session 2 Wednesday 12th June 13:00 - 15:00
  + Using diffrential expressed gene-lists, perform protein-protein interactions or gene-regulatory networks.
  + Manipulating and filtering gene-lists, pathways and clustered modules.
  + Creating sub-networks of specific derived clustered modules.
  + Enriching gene-list into pathways using FunctionalEnrichment - App.
- Session 3 Thursday 12th June 15:00 - 16:30
  + Pathway clustering using ClueGO - Functional Enrichment
  + Using ClueGOâ€™s features for integrating and querying multiple pathway database sources.
  + Use CluePedia (sister-software for ClueGO) to map gene regulatory information within specific pathways.
  
  