---
layout: page
organizer: Sheffield Bioinformatics Core
title: Crash Course in R and Prostate Cancer Bioinformatics
excerpt: "RNA-seq in R"
startDate: 2020-02-18
endDate: 2020-02-20
startTime: 9.30
endTime: 17.00
city: Oxford
postcode: 
country: United Kingdom
venue: 
description: "In this workshop, you will be learning how to analyse RNA-seq count data, using R. This will include reading the data into R, quality control and performing differential expression analysis and gene set testing, with a focus on the edgeR analysis workflow. You will learn how to generate common plots for analysis and visualisation of gene expression data, such as boxplots and heatmaps. You will also be learning how alignment and counting of raw RNA-seq data can be performed in R. This workshop is aimed at biologists interested in learning how to perform differential expression analysis of RNA-seq data when reference genomes are available.."
keywords: bioinformatics, NGS, RNA-seq, R
difficulty: beginner
contact: bioinformatics-core@sheffield.ac.uk
---

- University of Oxford - 18th - 20th February 2020
- 09:30am - 5pm
- Venue to be confirmed


![](/images/NDS_Ox_logo_v4.png)

## Overview

In this workshop, you will be learning how to analyse RNA-seq count data, using R. This will include reading the data into R, quality control and performing differential expression analysis and gene set testing, with a focus on the DESEq2 analysis workflow. You will learn how to generate common plots for analysis and visualisation of gene expression data, such as boxplots and heatmaps. You will also be learning how alignment and counting of raw RNA-seq data can be performed in R. This workshop is aimed at biologists interested in learning how to perform differential expression analysis of RNA-seq data when reference genomes are available.

## R packages

- For Day 2, please copy and paste [this R script](https://raw.githubusercontent.com/sheffield-bioinformatics-core/prostate-bioinformatics/master/install_bioc-packages.R?token=AA462NKRWTCYUUW2VQ3TKQ26JQD2E) into an R console to install the packages required for the workshop

## Course Data

- [Day 1](http://sbc.shef.ac.uk/workshops/2019-09-17-r/CourseData.zip)
- [Day 2](https://drive.google.com/file/d/1k-ol5waIdcayaCcyTLvhEYcMMWKiQjnx/view?usp=sharing)
- [Day 3]()

## Course Materials

Day 1

- [Course Overview](https://sbc.shef.ac.uk/prostate-bioinformatics/intro_slides.html)
- 09:30 - 11:00 [R basics](https://sbc.shef.ac.uk/workshops/2019-09-17-r/crash-course.nb.html#r_basics)
- 11:00 - 12:30 [Dealing with Data](https://sbc.shef.ac.uk/workshops/2019-09-17-r/crash-course.nb.html#dealing-with-data)
- 12:30 - 13:30 LUNCH (not provided)
- 13:30 - 14:30 Dealing with Data (continued)
- 14:30 - 16:00 [Plotting](https://sbc.shef.ac.uk/workshops/2019-09-17-r/crash-course.nb.html#plotting)
- 16:00 - 17:00 [Summarising and joining](https://sbc.shef.ac.uk/workshops/2019-09-17-r/crash-course.nb.html#summarising_and_grouping_with_dplyr)

Day 2

- 09:30 - 11:00 [Quality assessment and pre-processing for RNA-seq](https://sbc.shef.ac.uk/prostate-bioinformatics/rna-seq-preprocessing.nb.html)
- 11:00 - 12:30 [Differential Expression](https://sbc.shef.ac.uk/prostate-bioinformatics/rna-seq-de.nb.html)
- 12:30 - 13:30 LUNCH
- 13:30 - 15:30 [Annotation and Visualisation of RNA-seq results](https://sbc.shef.ac.uk/prostate-bioinformatics/rna-seq-annotation-visualisation.nb.html)
- 15:30 - 17:00 [Gene set testing](https://sbc.shef.ac.uk/prostate-bioinformatics/rna-seq-gene-set-testing.nb.html)

Day 3
- [Gene set testing notes](https://sbc.shef.ac.uk/prostate-bioinformatics/Day3.nb.html)
- 09:30 - 11:15: Basic Medical Statistics and Survival Analysis in R
- 11:15 - 11:30: Single Cell RNA seq and Spatial Transcriptomics workflows in R "Teaser"
- Coffee and Tea
- 11:30 - 13:30: "Shiny apps" for Prostate Cancer
- 13:30 - 14:30: LUNCH
- 14:30 - 17:00: Bring your own data workshop (Optional)

## Extra reading / tutorials

Other tutorials that we would recommend for further reading / practice

- [Data Organisation in Spreadsheets](https://datacarpentry.org/spreadsheet-ecology-lesson/)
- [Data Cleaining in OpenRefine](https://datacarpentry.org/OpenRefine-ecology-lesson/)
- [Github Introduction](https://guides.github.com/activities/hello-world/)
- [Seurat tutorials and walkthroughs](https://satijalab.org/seurat/vignettes.html)(for single-cell data)

## Who should attend this course?

Researchers in life sciences who want to get an appreciation for the computational steps involved in analysis RNA-seq data from Prostate cancer, and how to execute best-practice RNA-seq workflows in R.

## Objectives:- After this course you should be able to:

- Design properly your RNA-Seq experiments considering advantages and limitations of RNA-seq assays
- Assess the quality of your datasets
- Know what tools are available in Bioconductor for RNA-seq data analysis and understand the basic object-types that are utilised
- Produce a list of differentially expressed genes from an RNA-seq experiment 

## Aims:- During this course you will learn about:

- RNA sequencing technology and considerations on experiment design
- Sources of variation in RNA-seq data
- Differential expression analysis using edgeR and DEseq2
- Annotation resources in Bioconductor
- Identifying over-represented gene sets among a list of differentially expressed genes 
- Techniques for analysing spatial transcriptomics data in R
    
## Software installation
You will need to bring an internet-enabled laptop to the course and install the latest versions of both R ***and*** RStudio before coming to the course

### Windows

Install R by downloading and running [this .exe](http://cran.r-project.org/bin/windows/base/release.htm) file from CRAN. Also, please download and run the [RStudio installer for Windows](https://www.rstudio.com/products/rstudio/download/#download). Note that if you have separate user and admin accounts, you should run the installers as administrator (right-click on .exe file and select “Run as administrator” instead of double-clicking). Otherwise problems may occur later, for example when installing R packages.

### Mac

Install R by downloading and running [this .pkg](http://cran.r-project.org/bin/macosx/R-latest.pkg) file from CRAN. Also, please download and run [the RStudio installer for Mac](https://www.rstudio.com/products/rstudio/download/#download)

### Linux

You can download the binary files for your distribution from CRAN. Or you can use your package manager (e.g. for Debian/Ubuntu run `sudo apt-get install r-base` and for Fedora run `sudo yum install R`). Also, please download and run the [RStudio installer](https://www.rstudio.com/products/rstudio/download/#download).


## Instructors

- Dr. Mark Dunning, Sheffield Bioinformatics Core Director
- Dr. Andrew Erickson, Postdoc, University of Oxford

## Feedback



## Registration 

Registration is by invite only. Please see our [training page](https://sbc.shef.ac.uk/training) for other training opportunities, or contact us to discuss running a course at your location


## Acknowledgements

This course is based on the course [RNAseq analysis in R](http://combine-australia.github.io/2016-05-11-RNAseq/) originally prepared by [Combine Australia](https://combine.org.au/) and delivered on May 11/12th 2016 in Carlton. The course was then modified by Cancer Research Uk Cambridge Institute and delivered as part of the [CRUK Functional Genomics Autumn School](https://bioinformatics-core-shared-training.github.io/cruk-autumn-school-2017/)

![](https://raw.githubusercontent.com/sheffield-bioinformatics-core/RNAseq-R/gh-pages/images/combine_banner.png)

![](https://raw.githubusercontent.com/sheffield-bioinformatics-core/RNAseq-R/gh-pages/images/CRUK_Cambridge_Major%20Centre%20logo.jpg)
