---
layout: page
organizer: Sheffield Bioinformatics Core
title: Introduction to Genomic Variant Interpretation
excerpt: "Intro to Variants"
startDate: 2020-04-06
endDate: 2020-04-07
startTime: 9.30
endTime: 17.00
city: Sheffield
postcode: S3 7ND
country: United Kingdom
venue: Bartolome House, Seminar Room EG03
description: "This course covers state-of-the-art and best-practice tools for the analysis of genomes. We describe, and give hands-on experience of, the entire analysis workflow from raw data generated by a sequencing machine to deriving variant calls (e.g. Single Nucleotide Variants) that are ready for downstream analysis, interpretation and prioritisation. We will describe the steps involved to go from sequencing library to a prioritised, clinically-relevant list of DNA variants. Practical sessions will use the user-friendly Galaxy interface (https://usegalaxy.org/) to demonstrate tasks such as alignment, quality control, variant-calling and annotation."
keywords: bioinformatics, NGS, DNA-seq, variant-calling
difficulty: beginner
contact: bioinformatics-core@sheffield.ac.uk
---

- Sheffield - 6th/7th April 2020
- 09:30am - 5pm
- Bartolome House, Seminar Room EG03

<iframe src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d2379.712925739254!2d-1.4909138840310832!3d53.38418577957716!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x4879788327d13c2b%3A0x76151ebce3e59f6!2sBartolom%C3%A9+House%2C+Sheffield!5e0!3m2!1sen!2suk!4v1533301984260" width="600" height="450" frameborder="0" style="border:0" allowfullscreen></iframe>

## Registration 

Please book your place on the course using [this link](https://onlineshop.shef.ac.uk/conferences-and-events/faculty-of-medicine-dentistry-and-health/neuroscience/introduction-to-genomic-variant-interpretation-april-2020)

## Overview

Healthcare systems around the globe are transforming medical practice to incorporate genetic variation as a vital information used for diagnosis and treatment planning. Genome sequencing technologies allow us to detect all variants in a patient's genome and international collaborative efforts such as The 100,000 Genomes Project, The Cancer Genome Atlas (TCGA) and International Cancer Genome Consortium (ICGC) have begun to catalogue and release data on genomic variation in a variety of disease types.

However, such datasets pose new challenges in the way the data have to be analyzed, annotated and interpreted which are not trivial and are daunting to the clinician or biomedical scientist. This course covers state-of-the-art and best-practice tools for the analysis of genomes. We describe, and give hands-on experience of, the entire analysis workflow from raw data generated by a sequencing machine to deriving variant calls (e.g. Single Nucleotide Variants) that are ready for downstream analysis, interpretation and prioritisation.

We will describe the steps involved to go from sequencing library to a *prioritised*, *clinically-relevant* list of DNA variants. Practical sessions will use the user-friendly Galaxy interface (https://usegalaxy.org/) to demonstrate tasks such as *alignment*, *quality control*, *variant-calling* and *annotation*. 


## Who should attend this course?

Healthcare providers and researchers who want to get an appreciation for how variants are identified and interpreted, but not neccesarily working on this problem routinely. 

## Objectives:- After this course you should be able to:

- Recall the key pipeline steps in calling variants from NGS data
- Understand the contents of *fastq*, *bam* and *vcf* file formats for NGS data
- Execute key steps of the pipeline using the online Galaxy services presented
- Interpret a set of variant-calls using best-practice guidelines

## Aims:- During this course you will learn about:

- Common file formats for representing NGS data
- Aligning genome data to reference sequence using modern alignment programmes (e.g. *bwa*)
- How to determine the analytical sensitiviy of genomic tests
- Use of tools to call sequence variants (e.g. *GATK*) and annotation of variant call files using established databases
- Principles of integration of laboratory and clinical information, and place of best-practice guidelines for indicating the clinical significance of results (eg. *ACMG guidelines and standards*)

## Prerequisites

- No prior programming experience is required. Prior knowledge of NGS sequencing technologies would be an advantage, but not required.

## Instructors

- Dr Matthew Parker, Clinical Bioinformatics Core Scientist, Course Leader
- Dr Emily Chambers, Bioinformatics Core Scientist
- Dr Mark Dunning, Bioinformatics Core Director
- Dr Dennis Wang, Bioinformatics Lecturer
- Michael Spiller, Sheffield Childrens Hospital

## Timetable (provisional)

Day 1

- 09:30 - 09:40 [Introduction to course and Learning Objectives]()
- 09:40 - 10:15 [Overview of NGS]()
- 10:15 - 10:45 [Introduction to Galaxy]()
- 11:00 - 12:45 [Fastq files and alignment Practical]()
- 12:45 - 13:45 LUNCH (not provided)
- 13:45 - 14:30 Viewing Alignments in IGV
- 14:30 - 15:30 [Variant Calling and VCF files]()
- 15:30 - 17:00 [Variant Calling in Galaxy]()
- 17:00 [Day 1 Wrap-up]()

Day 2

- 09:30 - 09:40 [Overview of Day 1]()
- 09:40 - 10:45 [Variant Interpretation]()
- 11:00 - 12:30 [Variant Annotation]()
- 12:30 - 13:30 LUNCH (not provided)
- 13:30 - 15:30 [Variant Interpretation in a Clinical labroratory]()
- 15:30 - 17:00 [Case studies]()
- 17:00 [Day 2 and Course Wrap-up]()



